### PR TITLE
Feature 148 editor add option to adddeleteedit selectradio choices

### PR DESCRIPTION
**Pull request by btepu002xamk**
Added options to select/checkbox/radio fields and some table features.

### CODE REVIEW

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+          onChange={ event => updateFormSection({
+            ...section,
+            title: event.target.value
+          }) }
+        />
+      );
+    }
+
+    if (fieldIndex !== undefined && sectionIndex !== undefined) {
+      const field = pendingForm.sections![sectionIndex].fields![fieldIndex];
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:383]: Use destructing syntax

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+      if (!field) {
+        return null;
+      }
+      const disableForm = field.type === "select" || field.type === "date-time" || field.type === "radio" || field.type === "checklist" || field.type === "date";
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:387]: Rename this and check usage

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+          { renderFieldsCategoryTitle(strings.draftEditorScreen.editor.features.required) }
+          <FormControlLabel
+            label={ strings.generic.yes }
+            control={
+              <Checkbox
+                checked={ field.required }
+                onChange={ event => updateFormField({ ...field, required: event.target.checked }) }
+              />
+            }
+          />
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:414]: Split this to its own method

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+                <Checkbox
+                  checked
+                />
+              }
+            />
+            <Typography fontSize="12px">
+              { strings.draftEditorScreen.editor.features.selectableFieldsInfo }
+            </Typography>
+          </FormControl>
+        </>
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:433]: Split this to its own method

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+  }, [fieldIndex, sectionIndex]);
+
+  /**
+   * Component render
+   */
+  return (
+    <>
+      { renderFeatures() }
+    </>
+  );
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:452]:
```suggestion
  return renderFeatures();
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+          <FormControl disabled={ !disableForm }>
+            <FormControlLabel
+              label={ strings.generic.yes }
+              control={
+                <Checkbox
+                  checked
+                />
+              }
+            />
+            <Typography fontSize="12px">
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:429]: Use variant

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 7]
```diff
@@ -0,0 +1,455 @@
...
+        { title }
+      </Typography>
+    );
+  };
+
+  /**
+   * Update option text
+   * @param updateTextOption FieldOption text we are changing
+   * @param index index value
+   */
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:87]:
```suggestion
  /**
   * Update option text
   *
   * @param updateTextOption FieldOption text we are changing
   * @param index index value
   */
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 8]
```diff
@@ -0,0 +1,455 @@
...
+      draftForm.sections?.splice(sectionIndex, 1, metaformSection);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+  * Renders fields category title
+  * @param title Category title
+  */
+  const renderFieldsCategoryTitle = (title: string) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:72]: Is this method necessary? Naming is bit confusing, what category?

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/localization/en.json [hunk 9]
```diff
@@ -173,6 +174,9 @@
     "title": "Forms",
     "welcome": "Welcome to form portal",
     "welcomeInfo": "If you are a professional, you can log in. The visitor please return to the home page.",
+    "backToEssote": "Return to front page",
+    "welcome": "Welcome to form portal",
+    "welcomeInfo": "If you are a professional, you can log in. The visitor please return to the home page.",
```
[the remark this answers is not in the document]
Comment (btepu002xamk) [reply src/localization/en.json:179]: They are used in home screen.

File: src/components/editor/metaform-editor-right-drawer-feature.tsx (hunk 8 above)
Comment (btepu002xamk) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx:72]: Deleted whole method

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 10]
```diff
@@ -0,0 +1,473 @@
...
+   */
+  const addNewOptionField = () => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const optionsAmount = JSON.stringify(pendingForm.sections![sectionIndex].fields![fieldIndex].options?.length);
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.options?.push({ name: optionsAmount, text: "Valintakenttä" });
+    });
+    setPendingForm(updatedForm);
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:97]: Is this actually used with boolean?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 11]
```diff
@@ -0,0 +1,473 @@
...
+  /**
+   * Add new Radio / Boolean / Select option field
+   */
+  const addNewOptionField = () => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const optionsAmount = JSON.stringify(pendingForm.sections![sectionIndex].fields![fieldIndex].options?.length);
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.options?.push({ name: optionsAmount, text: "Valintakenttä" });
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:95]: Use localized value in text or just empty string

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 12]
```diff
@@ -0,0 +1,473 @@
...
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.options?.splice(index, 1, updateTextOption);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Add new Radio / Boolean / Select option field
+   */
+  const addNewOptionField = () => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:89]: Check naming, is this field or just field options?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 13]
```diff
@@ -0,0 +1,473 @@
...
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Delete optionfield
+   * @param index index value of optionfield what we delete
+   * @param event event value
+   */
+  const deleteOptionField = (event: any, index: number) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:105]: Check naming, maybe deleteFieldOptions would be more descriptive.

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 14]
```diff
@@ -0,0 +1,473 @@
...
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.options?.push({ name: optionsAmount, text: "Valintakenttä" });
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Delete optionfield
+   * @param index index value of optionfield what we delete
+   * @param event event value
+   */
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:104]:
```suggestion
  /**
   * Delete optionfield
   *
   * @param index index value of optionfield what we delete
   * @param event event value
   */
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 15]
```diff
@@ -0,0 +1,473 @@
...
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.options?.splice(index, 1);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Update slider value
+   * @param eventValue Value of min or max slider value
+   * @param minMax Min or Max, depending which value we are changing
+   */
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:119]:
```suggestion
  /**
   * Update slider value
   *
   * @param eventValue Value of min or max slider value
   * @param minMax Min or Max, depending which value we are changing
   */
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 16]
```diff
@@ -0,0 +1,473 @@
...
+  const updateSliderValue = (eventValue:string, minMax: string) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:120]:
```suggestion
  const updateSliderValue = (eventValue: string, minMax: string) => {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 17]
```diff
@@ -0,0 +1,473 @@
...
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const metaFormField = pendingForm.sections![sectionIndex].fields![fieldIndex];
+    if (minMax === "min") {
+      const updateField = {
+        ...metaFormField,
+        min: eventValue as unknown as number
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:128]:
```suggestion
        min: Number(eventValue)
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 18]
```diff
@@ -0,0 +1,473 @@
...
+      };
+      const updatedForm = produce(pendingForm, draftForm => {
+        draftForm.sections?.[sectionIndex]?.fields?.splice(fieldIndex, 1, updateField);
+      });
+      setPendingForm(updatedForm);
+    }
+    if (minMax === "max") {
+      const updateField = {
+        ...metaFormField,
+        max: eventValue as unknown as number
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:138]:
```suggestion
        max: Number(eventValue)
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 19]
```diff
@@ -0,0 +1,473 @@
...
+      });
+      setPendingForm(updatedForm);
+    }
+  };
+
+  /**
+   * Delete column
+   * @param index indexvalue of current column we are deleting
+   * @param event event value
+   */
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:151]:
```suggestion
  /**
   * Delete column
   *
   * @param index indexvalue of current column we are deleting
   * @param event event value
   */
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 20]
```diff
@@ -0,0 +1,473 @@
...
+  const deleteColumn = (event: any, index : number) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:152]:
```suggestion
  const deleteColumn = (event: any, index: number) => {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 21]
```diff
@@ -0,0 +1,473 @@
...
+   */
+  const addNewColumn = () => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const columnsAmount = JSON.stringify(pendingForm.sections![sectionIndex].fields![fieldIndex].columns?.length);
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.columns?.push({
+        type: columnType as MetaformTableColumnType,
+        name: columnsAmount,
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:173]: This should be always unique, is it?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 22]
```diff
@@ -0,0 +1,473 @@
...
+        title: `Column title ${columnsAmount}`
+      });
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Add custom html code
+   * @param htmlField html field
+   */
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:183]:
```suggestion
  /**
   * Add custom html code
   *
   * @param htmlField html field
   */
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 23]
```diff
@@ -0,0 +1,473 @@
...
+  const updateCustomHtml = (htmlField : MetaformField) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:184]:
```suggestion
  const updateCustomHtml = (htmlField: MetaformField) => {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 24]
```diff
@@ -0,0 +1,473 @@
...
+              type="number"
+              sx={{ height: "50px" }}
+              label={`${selectedType.title} max value`}
+              value={ selectedType.max}
+              onChange={ event => updateSliderValue(event.target.value, "max") }
+            />
+          </>
+        );
+      }
+      if (selectedType.type === "checklist" || selectedType.type === "radio" || selectedType.type === "select") {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:222]:
```suggestion
      }

      if (selectedType.type === "checklist" || selectedType.type === "radio" || selectedType.type === "select") {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 25]
```diff
@@ -0,0 +1,473 @@
...
+                        display: "flex",
+                        alignItems: "center",
+                        justifyContent: "center"
+                      }}
+                    />
+                  </Button>
+                </Stack>
+              );
+            })}
+            <Button fullWidth sx={{ height: "50px" }} onClick={ addNewOptionField }>{ strings.draftEditorScreen.editor.features.addSelectionField }</Button>
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:252]: Moves parameters to separate lines.

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/components/editor/metaform-editor-right-drawer-feature.tsx (hunk 10 above)
Comment (btepu002xamk) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx:97]: No, only checklist, radio and select

File: src/localization/fi.json [hunk 26]
```diff
@@ -224,7 +225,13 @@
...
         "sectionTitle": "Osion otsikko",
         "addSelectionField": "Lisää kenttä",
-        "addNewColumn": "Lisää uusi sarake"
+        "addNewColumn": "Lisää uusi sarake",
+        "addColumnType": "Anna kentän tyyppi",
+        "columnTextType": "Tekstikenttä",
+        "columnNumberType": "Numerokenttä",
+        "textOfSubmitButton": "Painikkeen teksti",
+        "newFieldOption":"Valintakenttä",
+        "addCustomHtml":"Lisää tähän kustomoitua html"
```
Comment (EVP1989) [inline src/localization/fi.json:234]:
```suggestion
        "addCustomHtml":"Lisää tähän HTML-sisältöä"
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 27]
```diff
@@ -0,0 +1,569 @@
...
+ * Draft editor right drawer feature component
+ */
+const MetaformEditorRightDrawerFeatureComponent: FC<Props> = ({
+  sectionIndex,
+  fieldIndex,
+  pendingForm,
+  setPendingForm
+}) => {
+  const [ metaformSectionTitle, setMetaformSectionTitle ] = React.useState<string | undefined>("");
+  const [ columnType, setColumnType ] = React.useState<string>("");
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:29]: Move useStates to React import

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 28]
```diff
@@ -0,0 +1,569 @@
...
+      draftForm.sections?.splice(sectionIndex, 1, metaformSection);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Update option text
+   * 
+   * @param updateTextOption FieldOption text we are changing
+   * @param index index value
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:74]: Rename index to more describing, maybe optionIndex?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 29]
```diff
@@ -0,0 +1,569 @@
...
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.options?.push({
+        name: `${strings.draftEditorScreen.editor.features.newFieldOption}-${optionsAmount}`,
+        text: `${strings.draftEditorScreen.editor.features.newFieldOption}-${optionsAmount}`
+      });
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Delete optionfield
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:104]:
```suggestion
   * Delete option field
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 30]
```diff
@@ -0,0 +1,569 @@
...
+   * 
+   * @param index index value of optionfield what we delete
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:106]: option index

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 31]
```diff
@@ -0,0 +1,569 @@
...
+   * @param event event value
+   */
+  const deleteFieldOptions = (event: any, index: number) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:109]: Why event is any?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 32]
```diff
@@ -0,0 +1,569 @@
...
+    }
+  };
+
+  /**
+   * Delete column
+   * 
+   * @param index indexvalue of current column we are deleting
+   * @param event event value
+   */
+  const deleteColumn = (event: any, index: number) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:158]: Why event is type any?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 33]
```diff
@@ -0,0 +1,569 @@
...
+        draftForm.sections?.[sectionIndex]?.fields?.splice(fieldIndex, 1, updateField);
+      });
+      setPendingForm(updatedForm);
+    }
+  };
+
+  /**
+   * Delete column
+   * 
+   * @param index indexvalue of current column we are deleting
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:155]: Column index?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 34]
```diff
@@ -0,0 +1,569 @@
...
+        type: columnType as MetaformTableColumnType,
+        name: columnsAmount,
+        title: columnsAmount
+      });
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Add custom html code in field
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:187]:
```suggestion
   * Add html code in field
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 35]
```diff
@@ -0,0 +1,569 @@
...
+   * 
+   * @param htmlField html field
+   */
+  const updateCustomHtml = (htmlField: MetaformField) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:191]:
```suggestion
  const updateHtml = (htmlField: MetaformField) => {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 36]
```diff
@@ -0,0 +1,569 @@
...
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Render slider scope values
+   *  
+   * @param selectedField selected metaformField
+   * */
+  const renderSliderScopeValues = (selectedField: MetaformField) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:206]:
```suggestion
   * @param selectedField selected metaformField
   */
  const renderSliderScopeValues = (selectedField: MetaformField) => {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 37]
```diff
@@ -0,0 +1,569 @@
...
+  const renderHtmlEditor = (selectedField: MetaformField) => {
+    const { html } = selectedField;
+    return (
+      <TextField
+        fullWidth
+        placeholder={ strings.draftEditorScreen.editor.features.addCustomHtml }
+        multiline
+        rows={ 4 }
+        value={ html }
+        onChange={ event => updateCustomHtml({ ...selectedField, html: event.target.value }) }
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:304]:
```suggestion
        onChange={ event => updateHtml({ ...selectedField, html: event.target.value }) }
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 38]
```diff
@@ -0,0 +1,569 @@
...
+      />
+    );
+  };
+
+  /**
+   * Update column title value
+   * 
+   * @param tableColumn MetaformTableColumn where we are changing title
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:312]:
```suggestion
   * @param tableColumn Metaform table column where we are changing title
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 39]
```diff
@@ -0,0 +1,569 @@
...
+   * @param index index value of current column title
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:313]: Column index? Rename also the parameter. Check for similar comments in whole file

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 40]
```diff
@@ -0,0 +1,569 @@
...
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.columns?.splice(index, 1, tableColumn);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Render features for adding, editing and deleting columns in table field
+   */
+  const renderTableColumns = () => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:328]:
```suggestion
  const renderTableColumnFeatures = () => {
```

Comment (EVP1989) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx:328]: Or maybe it should be renderTableFeatures

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 41]
```diff
@@ -0,0 +1,569 @@
...
+      }
+    }
+  };
+  
+  /**
+   * Render define user group component
+   * 
+   *  @param field selected field
+   */
+  const defineUserGroupComponent = (field: MetaformField) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:438]:
```suggestion
  const renderDefineUserGroup = (field: MetaformField) => {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 42]
```diff
@@ -0,0 +1,569 @@
...
+      </>
+    );
+  };
+
+  /**
+   * Render fieldTitle and Required or not component
+   * 
+   * @param field current field
+   */
+  const fieldTitleAndRequiredComponent = (field: MetaformField) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:468]:
```suggestion
  const renderFieldTitleAndRequired = (field: MetaformField) => {
```

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 43]
```diff
@@ -0,0 +1,567 @@
...
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Add custom html code in field
+   * 
+   * @param htmlField html field
+   */
+  const updateHtml = (htmlField: MetaformField) => {
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:189]:
```suggestion
  const updateHtmlField = (htmlField: MetaformField) => {
```

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 44]
```diff
@@ -0,0 +1,567 @@
...
+   */
+  const updateColumnTitle = (tableColumn: MetaformTableColumn, columnIndex: number) => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.[fieldIndex]?.columns?.splice(columnIndex, 1, tableColumn);
+    });
+    setPendingForm(updatedForm);
+  };
```
Comment (EVP1989) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:321]: Move this up, put render methods at the bottom of the component, before component return.

**Review by EVP1989 — COMMENTED** [review]
Added comments

**Review by EVP1989 — APPROVED** [review]

**Pull request merged**